src/test/java/com/techsophy/tsf/wrapperservice/CommentServiceTest.java

```diff
@@ -114,4 +86,58 @@ void getComments() throws Exception
         }
     }
 
+    @Test
+    void createCommentTest(){
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        ApiResponse apiResponse = Mockito.mock(ApiResponse.class);
+        CommentDTO commentDTO = new CommentDTO("taskId", "pid", "bk", "comment");
+        Mockito.when(restTemplate.exchange(anyString(), any(HttpMethod.class), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.ACCEPTED);
+        Mockito.when(objectMapper.convertValue(any(), eq(ApiResponse.class))).thenReturn(apiResponse);
+
+        Map<String, Object> actualOutput = commentService.createComment(commentDTO);
+        Map<String, Object> expectedOutput = objectMapper.convertValue(apiResponse.getData(), Map.class);
+        Assertions.assertEquals(expectedOutput, actualOutput);
+    }
+
+    @Test
+    void createCommentTestWhileThrowingException(){
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        CommentDTO commentDTO = new CommentDTO(null, "pid", null, "comment");
+        Mockito.when(restTemplate.exchange(anyString(), any(HttpMethod.class), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.BAD_REQUEST);
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> commentService.createComment(commentDTO));
+    }
+
+    @Test
+    void createCommentTestWhileCommentDtoIsNull(){
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        TaskModel task = Mockito.mock(TaskModel.class);
+        CommentDTO commentDTO = new CommentDTO(null, null, "bKey", "comment");
+        PaginationDTO<List<TaskInstanceDTO>> page = new PaginationDTO<>(List.of(taskInstanceDTO), 1, 1, 1, 1l, 1l);
+        Mockito.when(restTemplate.exchange(anyString(), any(HttpMethod.class), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.BAD_REQUEST);
+        Mockito.when(processService.getTasksByQuery(any(), any(), any())).thenReturn(page);
+        Mockito.when(objectMapper.convertValue(any(), any(TypeReference.class))).thenReturn(List.of(task));
+
+        Map<String, Object> actualOutput = commentService.createComment(commentDTO);
+        Map<String, Object> expectedOutput = objectMapper.convertValue(response, Map.class);
```

Comment:
Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.BAD_REQUEST); 
Map<String, Object> expectedOutput = objectMapper.convertValue(response, Map.class);

should be on line 115 to make it more readable.